app/assets/javascripts/discourse/app/components/share-popup.js

```diff
@@ -14,9 +14,12 @@ export default Component.extend({
   link: null,
   visible: null,
 
-  @discourseComputed
+  @discourseComputed("topic")
```

Comment:
Should we compute on topic.isPrivateMessage instead of just topic here?

Also why use `@discourseComputed` if we call this.topic directly?

Reply:
👍 computing on `topic.isPrivateMessage` is cleaner. 

To the second question, we need to compute this, otherwise when navigating from public to private topic this won't get recalculated.